src/mono/wasm/runtime/CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ target_link_libraries(dotnet
     ${MONO_ARTIFACTS_DIR}/libmono-component-hot_reload-static.a
     ${MONO_ARTIFACTS_DIR}/libmono-component-debugger-static.a
     ${MONO_ARTIFACTS_DIR}/libmono-component-diagnostics_tracing-stub-static.a
+    ${MONO_ARTIFACTS_DIR}/libmono-component-marshal-ilgen-static.a
     ${MONO_ARTIFACTS_DIR}/libmono-ee-interp.a
     ${MONO_ARTIFACTS_DIR}/libmonosgen-2.0.a
     ${MONO_ARTIFACTS_DIR}/libmono-ilgen.a
```

Comment:
interesting. Checked with Zoltan: wasm now always needs ilgen because it always uses the interpreter.
So it's only ios that really has a mode where they don't link in `libmono-ilgen.a`